src/components/MenuPanel/OrganizationSwitcher/Favorites.js

```diff
@@ -0,0 +1,148 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { FavoriteDaoType } from '../../../prop-types'
+import FavoriteRow from './FavoriteRow'
+
+class Favorites extends React.Component {
+  static propTypes = {
+    favoriteDaos: PropTypes.arrayOf(FavoriteDaoType),
+    currentDao: FavoriteDaoType,
+    onRequestClose: PropTypes.func.isRequired,
+    onDone: PropTypes.func.isRequired,
+  }
+
+  state = { localDaos: [] }
+
+  componentDidMount() {
+    this.setState({ localDaos: this.getLocalDaos() })
+  }
+
+  componentWillUnmount() {
+    const { onDone } = this.props
+    const { localDaos } = this.state
+    onDone(
+      localDaos.filter(dao => dao.favorited).map(dao => ({
+        name: dao.name,
+        address: dao.address,
+      }))
+    )
+  }
+
+  // Build the local DAO list based on the favorites. The favorite state is not
+  // directly reflected in the popup, to let users favorite / unfavorite items
+  // without seeing them being immediatly removed. For this reason, we need to
+  // maintain a separate state.
+  getLocalDaos() {
+    const { currentDao, favoriteDaos } = this.props
+
+    const localDaos = [
+      ...favoriteDaos
+        .map(dao => ({ ...dao, favorited: true }))
+        .sort(dao => (dao.address === currentDao.address ? -1 : 1)),
+    ]
+
+    // Do not add the current DAO if it is already in the favorites list
+    if (this.isDaoFavorited(currentDao)) {
+      return localDaos
+    }
+
+    return [
+      { ...currentDao, favorited: this.isDaoFavorited(currentDao) },
+      ...localDaos,
+    ]
+  }
+
+  isDaoFavorited({ address }) {
+    const { favoriteDaos } = this.props
+    return favoriteDaos.findIndex(dao => dao.address === address) > -1
```

Comment:
May I suggest to use `find` instead of `findIndex`?

Reply:
I generally use `.findIndex()` rather than `.find()` for this task, because `null` (returned by `.find()`) could be a valid entry in an array, while `-1` always means the same thing.

But it is kind of an old habit, following the `x.indexOf(y) > -1` pattern: what I should really use here is the [`some()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) method which is made for that 😄

About `.find()`, is it a preference or do you see a specific reason to use it here?

Reply:
TIL I should use `some` instead of `find` 😄